### ISSUE TITLE
Remaining time variable enhancements

### ISSUE THREAD

**Issue by CoastIT** [bitfocus/companion-module-obs-studio#144]
Hi,

I while ago I've coded a remaining time feature, meanwhile a similar feature was added to the  master. I'm now about to integrate my solution.

a) A feedback to let a button blink (1 sec feedback on / feedback off) when the remaining time is below a configureable threshold.

![Remaining_Time_Feedback_01](https://user-images.githubusercontent.com/99419948/154838872-8351d593-93e6-4285-95d1-f9cf298ce238.png)

Just my personal preference:
b) The remaining time variable shall be empty when a source is not playing
c) I would like to have an option to not display the "-" in front of the remaining time

![Remaining_Time_Feedback_02](https://user-images.githubusercontent.com/99419948/154839031-beca7d2d-14ee-4be1-90e3-ef0c3cea7d98.png)

For a) it's easy. There are some adaptions required to make the self.mediaSources datastructure persistant to be able to retrieve the values in a feedback. I've already done a POC on this.

For b) and c) I could introduce a new variable with another format (e.g. "media_time_remaining_2_Source Name") but this would increase the number of variables and might become confusing. The other idea I have in mind is to introduce a global setting which defines the format of the remaining time variable but this would not allow to use either formats in parallel. I tend to follow the first approach.

Any opinions on this?

**Comment by bryce-seifert** [comment]
> a) A feedback to let a button blink (1 sec feedback on / feedback off) when the remaining time is below a configureable threshold.

This sounds good, although I think it might be worth having the option to be a constant color or blink in case people want to use it with triggers.

> b) The remaining time variable shall be empty when a source is not playing

Would this be when the source is stopped, or also when the source is paused? I think there's some value/use cases for still displaying a time when the source is paused.

> c) I would like to have an option to not display the "-" in front of the remaining time

The reason this is currently there is it matches how elapsed/remaining time are displaying within OBS itself.

> For b) and c) I could introduce a new variable with another format (e.g. "media_time_remaining_2_Source Name") but this would increase the number of variables and might become confusing. The other idea I have in mind is to introduce a global setting which defines the format of the remaining time variable but this would not allow to use either formats in parallel. I tend to follow the first approach.

I don't think we should have two variables doing essentially the same thing. The global setting could work, but it seems like a minor cosmetic change to have a whole setting for. Something I think could justify this is more substantial view options, like only MM:SS format, or using seconds only, although I'm not sure if that's generally useful.

**Comment by CoastIT** [comment]
Thanks for your opinions. I'll start with the feedback stuff and then think about the time format topic. 

> This sounds good, although I think it might be worth having the option to be a constant color or blink in case people want to use it with triggers.

It looks like this is not possible as only boolean feedbacks can be used for triggers. As long as there is no blink support in the core I have to do that blinking in the feedback which means that the feedback is toggling on and off. I guess (not tested) this would also always trigger the triggers with each blink interval. I could of course create a non boolean feedback which would be always on but returns different colors but this would not help as such a feedback can not be used for triggers.

However, for a constant color feedback and also to trigger other stuff when the remaining time is below a threshold I think this might be already possible out of the box with an "internal: Check variable value" feedback (not tested).
